Dockerfile.terratest

```
@@ -0,0 +1,29 @@
+FROM golang:1.23
+
+# Create working directory and copy context into it
+WORKDIR /viya4-iac-azure
+COPY . .
```

It may be cleaner to have the user mount the full project with each invocation. That way they get their local changes automatically. It will make the command line invocation more complex, though.

Looking over the test doc, the suggestion is to have the user override the test subdir. That may be insufficient though, since the tests will be pulling file content from other parts of the project (e.g., `examples`). My gut feeling would be to avoid the `COPY` and instead mount the full project with `docker run`.

I haven't thought this one through deeply, so this comment really is more to ask the question and be sure it's been considered.

 If we stick with the `COPY`, I'd suggest moving it as late as possible in the Dockerfile. That way when folks have to rebuild (to update the image with their latest changes), the rebuild time will be quicker since the earlier layers can be reused.

---

That's a good suggestion. When we created this dockerfile, I had toyed with the idea of mounting the project rather than copying it. I ultimately decided against it because I was worried it would complicate the command line invocation.

But, after reading your thoughts, I agree that it's worth it. I hadn't considered that the user might want to mount more than the ./test directory. I was only thinking about new/updated tests, not new/updated features and fixes in the entire project. I'll go ahead and make those changes to the dockerfile and the doc.

---

@ephesused, I've updated the Dockerfile.terratest so it no longer copies the viya4-iac-azure directory. It now expects the user to mount it to the /viya4-iac-azure location. The entry point script will fail fast if it doesn't detect the mount. I've also updated any related doc around the change.
If that aligns with your suggestion, you can mark this as resolved. Thanks again for the input!